tests/python/pants_test/backend/jvm/tasks/jvm_compile/zinc/zinc_compile_integration_base.py

```diff
@@ -596,3 +596,72 @@ def test_hermetic(self):
             "examples/src/scala/org/pantsbuild/example/hello/exe", extra_args=extra_args
         ):
             pass
+
+    def test_differing_platforms(self):
+        with temporary_dir() as cache_dir, self.temporary_workdir() as workdir, temporary_dir(
+            root_dir=get_buildroot()
+        ) as src_dir:
+
+            config = {
+                "cache.compile.rsc": {"write_to": [cache_dir], "read_from": [cache_dir]},
+                "jvm-platform": {
+                    "platforms": {
+                        "six": {"source": "6", "target": "6"},
+                        "seven": {"source": "7", "target": "7"},
+                    },
+                    "default_platform": "six",
+                },
+            }
+
+            srcfile = os.path.join(src_dir, "org", "pantsbuild", "cachetest", "A.java")
+            buildfile = os.path.join(src_dir, "org", "pantsbuild", "cachetest", "BUILD")
+
+            self.create_file(
+                srcfile,
+                dedent(
+                    """
+                    package org.pantsbuild.cachetest;
+                    class A {
+                      public static void main(String[] args) {
+                        System.out.println("hello");
+                      }
+                    }
+                    """
+                ),
+            )
+            self.create_file(
+                buildfile,
+                dedent(
+                    """
+                    java_library(name='a',
+                                 sources=['A.java'])
+                    jvm_binary(name='bin',
+                               main='org.pantsbuild.cachetest.A',
+                               dependencies=[':a'])
+                    """
+                ),
+            )
+
+            cachetest_bin_spec = os.path.join(
+                os.path.basename(src_dir), "org", "pantsbuild", "cachetest:bin"
+            )
+            cachetest_spec = cachetest_bin_spec
+
+            # Cache values A.class
+            self.run_run(cachetest_spec, config, workdir)
```

Comment:
What is the expectation of this test? 

That the run passes, or are we checking certain behavior e.g. A.java is compiled with 6?

Also is the plan to have more internal test around 8/11 behaviors?

Reply:
The purpose of the test is to check that when you switch compile platforms, that the used platform changes.